packages/react-conformance/src/defaultTests.tsx

```diff
@@ -45,16 +47,20 @@ export const defaultTests: TestObject = {
    */
   'component-has-displayname': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
     const { Component } = testInfo;
-    if (Component.prototype.constructor) {
-      it(`constructor is a named function and matches displayName`, () => {
-        const constructorName = Component.prototype.constructor.name;
-        expect(constructorName).toEqual(testInfo.displayName);
-      });
-    } else {
-      it(`has a displayName`, () => {
-        expect(Component.displayName).toEqual(testInfo.displayName);
-      });
-    }
+
+    it(`has a displayName or constructor name`, () => {
+      const constructorName = Component.prototype?.constructor.name;
+      let displayName = Component.displayName || constructorName;
+
+      // This check is needed in case the Component is wrapped therefore the constructor's name is Wrapped,
+      // so we have to look at the display name and remove Base or Styled if the component was styled or is a base
```

Comment:
I'd go with something like this to be clear that the wrapping is using styled():
```suggestion
      // This check is needed in case the Component is wrapped with the v7 styled() helper, which returns a wrapper
      // component with constructor name Wrapped, and adds a Styled prefix to the displayName. Components passed to
      // styled() typically have Base in their name, so remove that too.
```